src/app/static/src/app/components/TreeSelect.vue

```diff
@@ -0,0 +1,57 @@
+<template>
+    <treeselect
+    :key="`${reRender}`"
+    :options="options"
+    :model-value="modelValue"
+    @update:model-value="input">
+        <slot></slot>
+    </treeselect>
+</template>
+<script lang="ts">
+    import { PropType, defineComponent } from "vue";
+    import Treeselect from "vue3-treeselect";
+
+    export default defineComponent({
+        components: {
+            Treeselect
```

Comment:
I wonder if it would be better to call this something more obviously different from the internal 3rd party component e.g. HintTreeSelect? `<treeselect>` and `<tree-select>` are pretty easily confused! (Probably not HintTreeSelect if we're going to pull this out into its own package, though not sure if this is on the cards..)

Reply:
agreed, I have gone with HintTreeSelect for now!